client/src/components/App.jsx

```diff
@@ -1,58 +1,93 @@
 import React from 'react';
+import axios from 'axios';
 import Reviews from './Reviews/Reviews.jsx';
 import ProductOverview from './ProductDetails/ProductOverview.jsx';
 import QuestionList from './QandA/QuestionList.jsx';
 import RelatedItems from './RelatedItems/RelatedItems.jsx';
 import NavBar from './Navbar.jsx';
 
-import Reviewsdata from './Reviews/reviewsData.js';
-import QuestionListdata from './QandA/QandAListData.js';
-import relatedItemsdata from './RelatedItems/relatedItemsData.js';
-import productListDummyData from './ProductDetails/sampleData/productListDummyData.js';
-
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      currentProductID: '39333',
+      currentProductID: '',
       currentStyleID: 234004,
-      productStylesArray: productListDummyData.productStyles.results,
-      currentProduct: productListDummyData.productToDisplay,
+      productStylesArray: [],
+      currentProduct: [],
       reviews: [],
-      questionList: QuestionListdata,
-      relatedItems: relatedItemsdata,
-
+      questionList: [],
+      relatedItems: [],
     };
+    this.cachedProducts = {};
     this.changeCurrentStyle = this.changeCurrentStyle.bind(this);
+  }
 
+  componentDidMount() {
+    this.getProductData('39333');
   }
 
+  getProductData(productId) {
```

Comment:
On first look, my thought would be to break this up into two helper functions, one for checking the cache and one for making the axios request, then just calling them here so as to reduce the amount of space taken up by this one functino (getProductData) -> but that's more of an optimisation perhaps.

Reply:
I might do this optimization later on Sam.